src/Parsers/ASTInsertQuery.cpp

```diff
@@ -48,11 +49,15 @@ void ASTInsertQuery::formatImpl(const FormatSettings & settings, FormatState & s
     }
     else
     {
+        if (infile)
+        {
+            settings.ostr << (settings.hilite ? hilite_keyword : "") << " FROM INFILE " << (settings.hilite ? hilite_none : "") << infile->as<ASTLiteral &>().value.safeGet<std::string>();
+        }
         if (!format.empty())
         {
             settings.ostr << (settings.hilite ? hilite_keyword : "") << " FORMAT " << (settings.hilite ? hilite_none : "") << format;
         }
-        else
+        else if (!infile)
```

Comment:
What will happen on queries like `INSERT INTO test FROM INFILE 'some_values' VALUES ('another', 'values')`? 
We should not allow queries with both `INFILE` and data for insert.

Reply:
Ok, I see that we will fail to parse such query